app/lib/malware_scan.rb

```diff
@@ -21,7 +18,7 @@ def call
 
     response = fetch_scan_result
 
-    if response.success?
+    if response.is_a?(Net::HTTPSuccess)
```

Comment:
`response` is been returned by `fetch_scan_result` which calls `blob_client`.

If we look at the gem's source code [AzureBlob::Client#get_blob](https://github.com/testdouble/azure-blob/blob/3b82d0c9129eb4ae4e9ad7911ed7b2a0732b1be9/lib/azure_blob/client.rb#L69) will return the result of [AzureBlob::Http#get](https://github.com/testdouble/azure-blob/blob/3b82d0c9129eb4ae4e9ad7911ed7b2a0732b1be9/lib/azure_blob/http.rb#L29) which can be 2 things:

1. The response body
2. An exception

I can see on line [33](https://github.com/DFE-Digital/register-trainee-teachers/pull/4494/files#diff-958d70781b30c9d0ee7a13f7b409d96fd213facedf14fd040c196dde20d6823dR33) you rescued however `response.is_a?(Net::HTTPSuccess)` and `response.body` don't look right since response should be a string blob and not an object.

Reply:
@Reyko I see what you mean, see [this commit](https://github.com/DFE-Digital/register-trainee-teachers/pull/4494/commits/aa65e24d4c255fb6c5c5440bd81ac4293a0d82ea). This treats the response just as a string object now and removes the check for `Net::HTTPSuccess`. I've also added `Net::HTTPClientError` to the rescue block, since the client might raise one of those as well. Let me know what you think, good spot 👍